File: src/main/java/com/fluxchess/pulse/Rank.java
/*
 * Copyright (C) 2013-2015 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */
package com.fluxchess.pulse;

final class Rank {

  static final int r1 = 0;
  static final int r2 = 1;
  static final int r3 = 2;
  static final int r4 = 3;
  static final int r5 = 4;
  static final int r6 = 5;
  static final int r7 = 6;
  static final int r8 = 7;

  static final int NORANK = 8;

  static final int[] values = {
      r1, r2, r3, r4, r5, r6, r7, r8
  };

  private Rank() {
  }

  static boolean isValid(int rank) {
    switch (rank) {
      case r1:
      case r2:
      case r3:
      case r4:
      case r5:
      case r6:
      case r7:
      case r8:
        return true;
      default:
        return false;
    }
  }

}

File: pulse-go/internal/pulse/rank.go
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package pulse

type Rank = int

const (
	_1 Rank = 0
	_2 Rank = 1
	_3 Rank = 2
	_4 Rank = 3
	_5 Rank = 4
	_6 Rank = 5
	_7 Rank = 6
	_8 Rank = 7

	NoRank Rank = 8
)

var (
	Ranks = [8]Rank{_1, _2, _3, _4, _5, _6, _7, _8}
)
